### PR TITLE
[Expression] Add DateTimeDiff prebuilt function

### PR DESCRIPTION
**Pull request by cosmicshuai**
fixes: #3933 

DateTimeDiff takes two timestamp string in ISO-8061 format, return a number of ticks that the two timestamp differs.

```
DateTimeDiff(Date1: string, Date2: string) => ticks: number
```

### CODE REVIEW

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 1]
```diff
@@ -3764,6 +3763,41 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
+                            {
+                                (dateTimeStart, error) = Ticks(ts1);
+                                if (error == null)
+                                {
+                                    (dateTimeEnd, error) = Ticks(ts2);
+                                }
+                            }
+                            else
+                            {
+                                error = $"{expr} can't evaluate.";
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3788]: >error = $"{expr} can't evaluate."; [](start = 32, length = 34)

Better error message would be "{expr} must have two ISO timestamps."

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 2]
```diff
@@ -3764,6 +3763,41 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
+                        IReadOnlyList<object> args;
+                        (args, error) = EvaluateChildren(expr, state, options);
+                        if (error == null)
+                        {
+                            if (args[0] is string ts1 && args[1] is string ts2)
+                            {
+                                (dateTimeStart, error) = Ticks(ts1);
+                                if (error == null)
+                                {
+                                    (dateTimeEnd, error) = Ticks(ts2);
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3783]: Missing error message if can't compute ticks.

Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs:3783]: This error is handling by the ExpressionFunctions.Ticks method. The error would happen when timestamp is not an ISO timestamp.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 1 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: error message updated

**Pull request merged**